Fabric/src/main/java/me/ferdz/placeableitems/block/component/impl/FragileBlockComponent.java

```diff
@@ -0,0 +1,97 @@
+package me.ferdz.placeableitems.block.component.impl;
+
+import java.util.function.Supplier;
+
+import me.ferdz.placeableitems.block.component.AbstractBlockComponent;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.Hand;
+import net.minecraft.util.hit.BlockHitResult;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+public class FragileBlockComponent extends AbstractBlockComponent {
+
+    private ContextualObjectCreator<ItemStack> itemStackCreator;
+    private ContextualObjectCreator<? extends Entity> entityCreator;
+
+    public FragileBlockComponent(ContextualObjectCreator<ItemStack> itemStackCreator, ContextualObjectCreator<? extends Entity> entityCreator) {
+        this.itemStackCreator = itemStackCreator;
+        this.entityCreator = entityCreator;
+    }
+
+    public FragileBlockComponent(Supplier<ItemStack> itemStackCreator, Supplier<? extends Entity> entityCreator) {
+        this(ContextualObjectCreator.fromSupplier(itemStackCreator), ContextualObjectCreator.fromSupplier(entityCreator));
+    }
+
+    public FragileBlockComponent(ContextualObjectCreator<ItemStack> itemStackCreator, Supplier<? extends Entity> entityCreator) {
+        this(itemStackCreator, ContextualObjectCreator.fromSupplier(entityCreator));
+    }
+
+    public FragileBlockComponent(Supplier<ItemStack> itemStackCreator, ContextualObjectCreator<? extends Entity> entityCreator) {
+        this(ContextualObjectCreator.fromSupplier(itemStackCreator), entityCreator);
+    }
+
+    public FragileBlockComponent() {
+        this((ContextualObjectCreator<ItemStack>) null, (ContextualObjectCreator<? extends Entity>) null);
+    }
+
+    @Override
+    public boolean onBlockActivated(BlockState state, World worldIn, BlockPos pos, PlayerEntity player, Hand handIn, BlockHitResult hit) throws NotImplementedException {
```

Comment:
I don't think the `throws` is necessary here

Reply:
This is due to my IDE wanting to override the parent, which _does_ throw this exception. Will remove it.